subgraphs/inverse-finance/src/common/getters.ts

```diff
@@ -0,0 +1,220 @@
+  
+    if (!protocol) {
+      protocol = new LendingProtocol(FACTORY_ADDRESS)
+      protocol.name = "Inverse Finance v1"
+      protocol.slug = "inverse-finance-v1"
+      protocol.schemaVersion = "1.1.0"
+      protocol.subgraphVersion = "1.1.0"
+      protocol.methodologyVersion = "1.1.0"
+      protocol.network = Network.ETHEREUM
+      protocol.type = ProtocolType.LENDING
+      ////// quantitative data //////
+      protocol.totalUniqueUsers = INT_ZERO
+      protocol.totalValueLockedUSD = BIGDECIMAL_ZERO
+      protocol.totalVolumeUSD = BIGDECIMAL_ZERO
+      protocol.totalDepositUSD = BIGDECIMAL_ZERO
+      protocol.totalBorrowUSD = BIGDECIMAL_ZERO
+      //protocol.usageMetrics
+      //protocol.financialMetrics
+      //protocol.markets
+      protocol.lendingType = LendingType.CDP
+      protocol.riskType = RiskType.GLOBAL
+      
+      protocol.save()
+    }  
+    return protocol
+  }
+
+  export function getOrCreateMarket(marketAddr: string,
+                                    event: ethereum.Event
+                                    ): Market {
+    
+    let market = Market.load(marketAddr)
+  
+    if ( market == null ) {
+      let contract = CErc20.bind(Address.fromString(marketAddr))
+      
+      let asset = ZERO_ADDRESS //default
+      let tryAsset = contract.try_underlying()
+      if (!tryAsset.reverted) {
+        let asset = tryAsset.value.toHexString()
+      }
+
+      market = new Market(marketAddr)
+      market.protocol = FACTORY_ADDRESS
+      market.inputTokens = [asset]
+      market.outputToken = marketAddr //Token.load(marketAddr).id
+      market.rewardTokens = [prefixID(RewardTokenType.DEPOSIT, INV_ADDRESS), 
+                             prefixID(RewardTokenType.BORROW, INV_ADDRESS)]
+
+      market.totalValueLockedUSD = BIGDECIMAL_ZERO
+      market.totalVolumeUSD = BIGDECIMAL_ZERO
+      market.totalDepositUSD = BIGDECIMAL_ZERO
+      market.totalBorrowUSD = BIGDECIMAL_ZERO
+      market.inputTokenBalances = [BIGINT_ZERO]
+      market.inputTokenPricesUSD = [BIGDECIMAL_ZERO]
+      market.outputTokenSupply = BIGINT_ZERO
+      market.outputTokenPriceUSD = BIGDECIMAL_ZERO
+      market.rewardTokenEmissionsAmount = [BIGINT_ZERO]
+      market.rewardTokenEmissionsUSD = [BIGDECIMAL_ZERO]
+
+      //market.snapshots
+      market.name = contract.name()
+      // markets[address].isListed never resets after set to true
+      //QQQ: Should borrowing/minting/transfering pauses considered inactive? 
+      market.isActive = true
+      market.canUseAsCollateral = false
+      market.canBorrowFrom = true //borrowGuardianPaused is default to false
+      market.maximumLTV = BIGDECIMAL_ZERO
+      market.liquidationThreshold = BIGDECIMAL_ZERO
+      market.liquidationPenalty = BIGDECIMAL_ZERO
+      market.depositRate = BIGDECIMAL_ZERO
+      //market.stableBorrowRate = BIGDECIMAL_ZERO
+      market.variableBorrowRate = BIGDECIMAL_ZERO
+      //market.deposits
```

Comment:
same here, add comments saying these are derived fields that don't need to be inialized